graph/GuidelinesGraph.md

```diff
@@ -0,0 +1,441 @@
+following goals:
+
+\- Developer friendly via consistent naming, patterns, and web standards (HTTP,
+REST, JSON)
+
+\- Work well with SDKs in many programming languages.
+
+\- Sustainable & evolvable via clear API contracts.
+
+The Microsoft Graph guidelines are an extension of the Microsoft REST API
+guidelines. Readers are assumed also be reading and following the Microsoft REST API
+guidelines except where this document outlines specific differences or exceptions to those guidelines.
+Together these guidelines and a library of API patterns serve as the means by
+which API teams discuss and come to consensus on API review requirements.
+
+Technology and software are constantly changing and evolving, and as such, this
+is intended to be a living document. API guidelines that change frequently lead to an uneven and inconsistent API surface. Consequently, this document will more frequently change to add guidance in areas previously uncovered, or to clarify existing guidance. It will less frequently change the directional guidance it has already provided. [Open an
+issue](https://github.com/microsoft/api-guidelines/issues/new/choose) to suggest
+a change or propose a new idea.
+
+### Legend
+
+This document offers prescriptive guidance labeled as follows:
+
+:heavy_check_mark: **MUST** satisfy this specification. 
+
+:no_entry: **MUST NOT** use this pattern. 
+
+:ballot_box_with_check: **SHOULD** fulfill this specification. 
+
+:warning: **SHOULD NOT** adopt this pattern. 
+
+If not following these advices, you MUST disclose your reasons during the Graph API review.
+
+## Design Approach
+
+The design of your API is arguably the most important investment you will make. API design is what creates the first impression for developers when they discover and learn how to use your APIs. We promote API-first design approach where you begin your product design by focusing on how information will be exchanged and represented and creating an interface contract for your API which is followed by design and implementation of the backing service. This approach ensures decoupling of the interface from your implementation and is essential for agility, predictability, and reuse of your APIs. Established interface contract allows developers to use your API while internal teams are still working on implementation, API specifications enable designing of user experience and test cases in parallel. Starting with user-facing contracts also promotes a good understanding of system interactions, your modeling domain, and understanding of how the service will evolve. Microsoft Graph supports resource and query-based API styles which follow HTTP, REST, and JSON standards, where API contract is described using ODATA conventions and schema definition (see Documentation · OData - the Best Way to REST).
+[Documentation · OData - the Best Way to REST](https://www.odata.org/documentation/)).
+
+In general API design includes the following steps:
```

Comment:
For me, one key design element is mapping end-user scenarios/use cases to request/response patterns to understand what the API consumer experience would be.  That doesn't seem to factor into this list though.

Reply:
Agree.

Reply:
Outline the key current and future scenarios for API consumers